assets/app/scripts/services/deployments.js

```diff
@@ -0,0 +1,257 @@
+          DataService.update("deploymentconfigs", deploymentConfigName, newDeploymentConfig, $scope).then(
+            function(rolledBackDeploymentConfig) {
+              $scope.alerts = $scope.alerts || {};
+              $scope.alerts["rollback"] = 
+                {
+                  type: "success",
+                  message: "Deployment #" + rolledBackDeploymentConfig.status.latestVersion + " is rolling back " + deploymentConfigName + " to " + deploymentName + ".",
+                };
+            },
+            function(result) {
+              $scope.alerts = $scope.alerts || {};
+              $scope.alerts["rollback"] = 
+                {
+                  type: "error",
+                  message: "An error occurred while rolling back the deployment.",
+                  details: $filter('getErrorDetails')(result)
+                };
+            }
+          );
+        },
+        function(result) {
+          $scope.alerts = $scope.alerts || {};          
+          $scope.alerts["rollback"] = 
+            {
+              type: "error",
+              message: "An error occurred while rolling back the deployment.",
+              details: $filter('getErrorDetails')(result)
+            };
+        }
+      );
+    };
+
+    DeploymentsService.prototype.cancelRunningDeployment = function(deployment, $scope) {
+      var deploymentName = deployment.metadata.name;
+      var deploymentConfigName = $filter('annotation')(deployment, 'deploymentConfig');        
+      var req = angular.copy(deployment);
+
+      // TODO: we need a "cancel" api endpoint so we don't have to do this manually
+
+      // set the cancellation annotations
+      var deploymentCancelledAnnotation = $filter('annotationName')('deploymentCancelled');
+      var deploymentStatusReasonAnnotation = $filter('annotationName')('deploymentStatusReason');
+      req.metadata.annotations[deploymentCancelledAnnotation] = "true";
+      req.metadata.annotations[deploymentStatusReasonAnnotation] = "The deployment was cancelled by the user";
+
+      // update the deployment with cancellation annotations
+      DataService.update("replicationcontrollers", deploymentName, req, $scope).then(
+        function() {
+          $scope.alerts = $scope.alerts || {};
+          $scope.alerts["cancel"] = 
+            {
+              type: "success",
+              message: "Cancelling deployment " + deploymentName + " of " + deploymentConfigName + ".",
+            };
+        },
+        function(result) {
+          $scope.alerts = $scope.alerts || {};
+          $scope.alerts["cancel"] = 
+            {
+              type: "error",
+              message: "An error occurred while cancelling the deployment.",
+              details: $filter('getErrorDetails')(result)
+            };
+        }
+      );
+    };
+
+    DeploymentsService.prototype.deploymentIsLatest = function(deploymentConfig, deployment) {
```

Comment:
It feels like some of these methods like deploymentIsLatest would be better as filters since they're used only in views.